src/System.Device.Gpio/System/Device/Gpio/Drivers/HummingBoardDriver.Linux.cs

```diff
@@ -1,126 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace System.Device.Gpio.Drivers
-{
-    public partial class HummingBoardDriver : GpioDriver
```

Comment:
For all of these, we should probably add a check in their constructor to make sure that they are being initialized in a supported platform, so that if somebody tries to initialize  a libgpioddriver in windows, we will still throw platformNotSupported.

Reply:
I agree. That part I wasn't so sure about in detail. There are several possibilities here:
1. Add check in .ctor (not very nice, since ctors throwing for reasons other than invalid arguments is not good style)
2. Provide a public `Initialize` method for this check (cleaner when it comes to error handling, but may lead to usage errors)
3. Deprecate the usage of the existing .ctors of these classes and put the test logic to the corresponding `Board` implementation. Then it needs to be done (and implemented) only once. 

I'd prefer to do 3, since the problem not only happens for all the GpioDrivers but also for all the I2C drivers, the PWM drivers, the SPI drivers, etc. I'd then do 1 just for backwards compatibility (and provide a new ctor when created from the Board).

Reply:
1) already works, at least if you're not so very peculiar about the exception type being thrown. (DllNotFoundException, NotSupportedException, PlatformNotSupportedException, throw the dice). 
I'll start with 3) but on a separate branch, otherwise there's probably to much stuff on the same PR in the end.

Reply:
Should be working as expected now